pkg/controllers/resources/ingresses/syncer.go

```diff
@@ -117,3 +133,82 @@ func (s *ingressSyncer) SyncToVirtual(ctx *synccontext.SyncContext, event *syncc
 
 	return patcher.CreateVirtualObject(ctx, event.Host, vIngress, s.EventRecorder(), true)
 }
+
+func (s *ingressSyncer) ExcludeVirtual(obj client.Object) bool {
+	ingress, ok := obj.(*networkingv1.Ingress)
+	if !ok ||
+		((ingress.Spec.IngressClassName == nil || *ingress.Spec.IngressClassName == "") && selector.IsLabelSelectorEmpty(s.labelSelector)) {
+		return false
+	}
+
+	ingressClass := &networkingv1.IngressClass{}
+	err := s.physicalClusterClient.Get(context.Background(), types.NamespacedName{Name: *ingress.Spec.IngressClassName}, ingressClass)
+	if err != nil {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Ingress %q will not be synced to host cluster, because IngressClass %q couldn't be found: %v", ingress.Name, *ingress.Spec.IngressClassName, err))
+		return true
+	}
+
+	exclude := !selector.StandardLabelSelectorMatches(ingressClass, s.labelSelector)
+	if exclude {
+		klog.FromContext(context.Background()).Info(
+			fmt.Sprintf("Warning: Ingress %q will not be synced to host cluster, because IngressClass %q does NOT match the label selector in the 'sync.fromHost.ingressClasses' configuration", ingress.Name, *ingress.Spec.IngressClassName))
+	}
+	return exclude
+}
+
+func (s *ingressSyncer) ExcludePhysical(_ client.Object) bool {
+	return false
+}
+
+func (s *ingressSyncer) ModifyController(registerCxt *synccontext.RegisterContext, builder *builder.Builder) (*builder.Builder, error) {
+	loggerDebug := func(verb, objectName string) {
+		klog.FromContext(registerCxt.Context).V(1).Info(
+			fmt.Sprintf("%s triggers requeue of ingresses related with ingressClass %q", verb, objectName))
+	}
+	eventHandler := handler.Funcs{
+		CreateFunc: func(_ context.Context, e event.CreateEvent, q workqueue.TypedRateLimitingInterface[ctrl.Request]) {
+			loggerDebug("creation", e.Object.GetName())
+			requeueRelatedIngresses(registerCxt, nil, e.Object, q)
+		},
+		UpdateFunc: func(_ context.Context, e event.UpdateEvent, q workqueue.TypedRateLimitingInterface[ctrl.Request]) {
+			loggerDebug("update", e.ObjectNew.GetName())
+			requeueRelatedIngresses(registerCxt, e.ObjectOld, e.ObjectNew, q)
+		},
+		DeleteFunc: func(_ context.Context, e event.DeleteEvent, q workqueue.TypedRateLimitingInterface[ctrl.Request]) {
+			loggerDebug("delete", e.Object.GetName())
+			requeueRelatedIngresses(registerCxt, e.Object, nil, q)
+		},
+	}
+
+	return builder.Watches(&networkingv1.IngressClass{}, eventHandler), nil
```

Comment:
Same here we don't have permissions! Also why is this needed in the first place?